server/server_test.go

```diff
@@ -106,6 +104,31 @@ func TestUnversionedGetHealthCheckBundleActivationSingle(t *testing.T) {
 	}
 }
 
+func TestUnversionedGetHealthCheckBundleActivationSingleLegacy(t *testing.T) {
+
+	// Initialize the server as if there is no bundle plugin
+
+	f := newFixture(t)
+
+	ctx := context.Background()
+
+	err := storage.Txn(ctx, f.server.store, storage.WriteParams, func(txn storage.Transaction) error {
```

Comment:
What does the health check report before this transaction? Do we have a test that covers that? I'd expect it to be unhealthy if ?bundle is defined.

Reply:
In this case it actually returns healthy because there is no known bundle. This test is somewhat synthetic to trigger the panic as it starts the server up without a bundle then fakes adding ones manifest in the store. 

Bundles loaded via the data paths are processed when the runtime starts up and before the server ever starts serving so it isn't possible to call the health check. In practice the end result is that calls to the health check fail because the server isn't running yet, and opa would exit with an error initializing the runtime.